packages/superset-ui-encodable/test/encoders/Encoder.test.ts

```diff
@@ -33,19 +44,157 @@ describe('Encoder', () => {
   });
   describe('.getChannelNames()', () => {
     it('returns an array of channel names', () => {
-      expect(encoder.getChannelNames()).toEqual(['x', 'y', 'color', 'shape', 'tooltip']);
+      expect(encoder.getChannelNames()).toEqual(['x', 'y', 'color', 'radius', 'shape', 'tooltip']);
     });
   });
   describe('.getChannelEncoders()', () => {
     it('returns an array of channel encoders', () => {
-      expect(encoder.getChannelEncoders()).toHaveLength(5);
+      expect(encoder.getChannelEncoders()).toHaveLength(6);
     });
   });
   describe('.getGroupBys()', () => {
     it('returns an array of groupby fields', () => {
       expect(encoder.getGroupBys()).toEqual(['brand', 'make', 'model']);
     });
   });
+  describe('.getLegendInformation()', () => {
+    it('returns information for each field', () => {
+      const legendInfo = factory
+        .create({
+          color: { type: 'nominal', field: 'brand', scale: { range: ['red', 'green', 'blue'] } },
+          shape: { type: 'nominal', field: 'brand', scale: { range: ['circle', 'diamond'] } },
+        })
+        .getLegendInformation([{ brand: 'Gucci' }, { brand: 'Prada' }]);
+
+      expect(stripFunction(legendInfo)).toEqual([
+        {
+          field: 'brand',
+          type: 'nominal',
+          items: [
+            {
+              input: 'Gucci',
+              output: {
+                color: 'red',
+                radius: 5,
+                shape: 'circle',
+              },
+            },
+            {
+              input: 'Prada',
+              output: {
+                color: 'green',
+                radius: 5,
+                shape: 'diamond',
+              },
+            },
+          ],
+        },
+      ]);
+    });
+    it('ignore channels that are ValueDef', () => {
+      const legendInfo = factory
+        .create({
+          color: { type: 'nominal', field: 'brand', scale: { range: ['red', 'green', 'blue'] } },
```

Comment:
should the input be a non-nominal type? (the above test tests this same thing to some extent right?)

Reply:
It has some similarity with the test above it, but this one focuses on the omitted channels (`shape` which defaults to `ValueDef` `{ value: 'circle' }`)